### ISSUE TITLE
Compose V2 doesn't work behind proxy.

### ISSUE THREAD

**Issue by JugeHuge** [docker/compose#9034]
Compose V2 doesn't follow configured server proxy information when trying to build.
This will cause build fail as it can not connect to outside for update image from different repos.
Using docker-compose works fine by following proxy configurations and connecting outside repos.

**Steps to reproduce the issue:**
1. Have server behind proxy to connect outside, docker installed and proxies configured. Compose V2 installed.
2. Build image with compose. There should actions that need i.e. fetch updates to image
3. Build fails as fetch fails to timouts.

**Describe the results you received:**
Step 5/9 : RUN apt-get update --fix-missing  && apt-get install -y nano xxd wget curl telnet net-tools dnsutils
 ---> Running in df436b538840

Err:1 http://deb.debian.org/debian bullseye InRelease
  Could not connect to debian.map.fastlydns.net:80 (151.101.86.132), connection timed out Unable to connect to deb.debian.org:http:
Err:2 http://deb.debian.org/debian bullseye-updates InRelease
  Unable to connect to deb.debian.org:http:

Err:3 http://security.debian.org/debian-security bullseye-security InRelease
  Could not connect to debian.map.fastlydns.net:80 (151.101.86.132), connection timed out Could not connect to security.debian.org:80 (151.101.194.132), connection timed out Could not connect to security.debian.org:80 (151.101.130.132), connection timed out Could not connect to security.debian.org:80 (151.101.66.132), connection timed out Could not connect to security.debian.org:80 (151.101.2.132), connection timed out
Reading package lists...
[91mW: Failed to fetch http://deb.debian.org/debian/dists/bullseye/InRelease  Could not connect to debian.map.fastlydns.net:80 (151.101.86.132), connection timed out Unable to connect to deb.debian.org:http:
W: Failed to fetch http://security.debian.org/debian-security/dists/bullseye-security/InRelease  Could not connect to debian.map.fastlydns.net:80 (151.101.86.132), connection timed out Could not connect to security.debian.org:80 (151.101.194.132), connection timed out Could not connect to security.debian.org:80 (151.101.130.132), connection timed out Could not connect to security.debian.org:80 (151.101.66.132), connection timed out Could not connect to security.debian.org:80 (151.101.2.132), connection timed out
W: Failed to fetch http://deb.debian.org/debian/dists/bullseye-updates/InRelease  Unable to connect to deb.debian.org:http:
W: Some index files failed to download. They have been ignored, or old ones used instead.

**Describe the results you expected:**
Step 5/9 : RUN apt-get update --fix-missing  && apt-get install -y nano xxd wget curl telnet net-tools dnsutils
---> Running in f1a6614f18ff
Get:1 http://security.debian.org/debian-security bullseye-security InRelease [44.1 kB]
Get:2 http://deb.debian.org/debian bullseye InRelease [116 kB]
Get:3 http://deb.debian.org/debian bullseye-updates InRelease [39.4 kB]
Get:4 http://security.debian.org/debian-security bullseye-security/main amd64 Packages [99.5 kB]
Get:5 http://deb.debian.org/debian bullseye/main amd64 Packages [8180 kB]
Get:6 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [2592 B]
Fetched 8481 kB in 2s (5237 kB/s)

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker compose version`:**

```
Docker Compose version v2.2.2
```
**Output of `docker-compose version`:**

```
docker-compose version 1.28.5, build c4eb3a1f
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Docker Buildx (Docker Inc., v0.7.1-docker)
  compose: Docker Compose (Docker Inc., v2.2.2)
  scan: Docker Scan (Docker Inc., v0.12.0)

Server:
 Containers: 42
  Running: 0
  Paused: 0
  Stopped: 42
 Images: 189
 Server Version: 20.10.12
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc version: v1.0.2-0-g52b36a2
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 4.15.0-163-generic
 Operating System: Ubuntu 18.04.6 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 6
 Total Memory: 7.747GiB
 Name: noname
 ID: KG6N:K5TB:MYHO:47YX:P4MW:RG2P:U5TG:P77P:GTUY:YEMV:7XJU:AFGQ
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http://abc.xyz.com:3128
 HTTPS Proxy: http://abc.xyz.com:3128
 No Proxy: 127.0.0.1,localhost
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  registry.device.management:5000
  127.0.0.0/8
 Registry Mirrors:
  https://xyz-abc.xyz.com/
 Live Restore Enabled: false

```

**Additional environment details:**

**Comment by fjf2002** [comment]
Any news here?

**Comment by JugeHuge** [comment]
> Any news here?

Tested v2.6.0 and still no go when behind proxy

**Comment by rfay** [comment]
I put together a little demo on this. It's clear that despite the (claimed? fix in 
* https://github.com/docker/compose-cli/pull/2053

it still doesn't work.

https://github.com/rfay/docker-compose-proxy shows the exact output, using the same Dockerfile with either `docker compose` 2.6.0 or `docker` client. 

Docker is able to build because HTTP_PROXY gets injected into the build. `docker compose` is not able to build because it's not injected in there. 

Please give it a try, it's easy to see. Just check out the repo and go through the 4 simple steps.

**Comment by JugeHuge** [comment]
I tested 2.11.2 version and it seemed to be working. Not sure if company has changed something on proxies.
Can @JacobHanimann @croensch or @rfay test if this is now working?

**Comment by rfay** [comment]
@JugeHuge did you walk through https://github.com/rfay/docker-compose-proxy  from https://github.com/docker/compose/issues/9034#issuecomment-1178418031 ?

**Comment by JacobHanimann** [comment]
> I tested version 2.11.2 and it seemed to be working. Not sure if company has changed something on proxies. Can @JacobHanimann @croensch or @rfay test if this is now working?

Version v2.11.2 did not work for me. I downgraded docker-compose to version 1.19.0 to make the build command work!

**Comment by JugeHuge** [comment]
> @JugeHuge did you walk through https://github.com/rfay/docker-compose-proxy from [#9034 (comment)](https://github.com/docker/compose/issues/9034#issuecomment-1178418031) ?

sorry for late reply, Yes i did.
I tested with 2.2.0, 2.6.0, 2,10,2, 2.11.0, 2.11.2 and 2.12.2
Something in 2.11.0 release fixed issue as previous 2.10.2 failed as the older ones.

**Comment by rfay** [comment]
Thanks @JugeHuge  - You're right. It didn't work with v2.10.2, but it does work now with 2.12.2.

I guess you can close this issue! Yay!
